praw/models/reddit/subreddit.py

```diff
@@ -2174,6 +2179,84 @@ def _upload_image(self, image_path, data):
                 raise APIException(error_type, error_value, None)
             return response
 
+    def _upload_style_asset(self, image_path, data):
+        data['filepath'] = basename(image_path)
+        data['mimetype'] = 'image/jpeg'
+        if image_path.lower().endswith('.png'):
+            data['mimetype'] = 'image/png'
+        url = API_PATH['style_asset_lease'].format(subreddit=self.subreddit)
+
+        # until we learn otherwise, assume this request always succeeds
+        upload_lease = self.subreddit._reddit.post(url, data=data)['s3UploadLease']
+        upload_data = {item['name']: item['value']
+                       for item in upload_lease['fields']}
+        upload_url = 'https:{}'.format(upload_lease['action'])
+
+        with open(image_path, 'rb') as image:
+            response = self.subreddit._reddit._core._requestor._http.post(
+                upload_url, data=upload_data, files={'file': image})
+        response.raise_for_status()
+
+        structured_style = {}
+        image_url = response.headers['location']
+        structured_style[data['imagetype']] = image_url
```

Comment:
Related to my above comment: This line assumes an `'imagetype'` key in `data`, which also makes me think having `image_type` explicitly as a parameter might be better.

Reply:
Agreed